enginetest/enginetests.go

```diff
@@ -2395,6 +2395,33 @@ func TestWindowAgg(t *testing.T, harness Harness) {
 		{4, 0.0},
 		{5, 0.0},
 	}, nil, nil)
+
+	TestQuery(t, harness, e, `SELECT a, first_value(b) over (partition by c order by b) FROM t1 order by a`, []sql.Row{
+		{0, 0},
+		{1, 1},
+		{2, 0},
+		{3, 0},
+		{4, 0},
+		{5, 0},
+	}, nil, nil)
+
+	TestQuery(t, harness, e, `SELECT a, first_value(a) over (partition by b) FROM t1 order by a`, []sql.Row{
+		{0, 0},
+		{1, 1},
+		{2, 2},
+		{3, 0},
+		{4, 1},
+		{5, 5},
+	}, nil, nil)
+
+	TestQuery(t, harness, e, `SELECT a, first_value(a-1) over (partition by b order by c) FROM t1 order by a`, []sql.Row{
```

Comment:
Can you add a test case that does a first value over another column that is not a?

Reply:
👍